src/Microsoft.AspNet.Mvc.Localization/LocalizedHtmlString.cs

```diff
@@ -41,11 +57,145 @@ public LocalizedHtmlString(string key, string value, bool isResourceNotFound)
         /// <summary>
         /// The string resource.
         /// </summary>
-        public string Value => ToString();
+        public string Value { get; }
 
         /// <summary>
         /// Gets a flag that indicates if the resource is not found.
         /// </summary>
         public bool IsResourceNotFound { get; }
+
+        /// <inheritdoc />
+        public void WriteTo(TextWriter writer, HtmlEncoder encoder)
+        {
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+
+            if (encoder == null)
+            {
+                throw new ArgumentNullException(nameof(encoder));
+            }
+
+            var htmlTextWriter = writer as HtmlTextWriter;
+            if (htmlTextWriter == null)
+            {
+                FormatValue(writer, encoder, Value, _arguments);
+            }
+            else
+            {
+                htmlTextWriter.Write(this);
+            }
+        }
+
+        private void FormatValue(
```

Comment:
Mostly copied from `HtmlLocalizer` but old code used a second `StringBuilder` rather than the current `TextWriter`. Also added `AppendCurlyBrace()`.